pkgs/tools/misc/opentelemetry-collector/contrib.nix

```diff
@@ -2,28 +2,45 @@
 , fetchFromGitHub
 , lib
 , stdenv
+, systemdMinimal
+, withSystemd ? false
 }:
 
 buildGoModule rec {
   pname = "opentelemetry-collector-contrib";
-  version = "0.66.0";
+  version = "0.76.1";
 
   src = fetchFromGitHub {
     owner = "open-telemetry";
     repo = "opentelemetry-collector-contrib";
     rev = "v${version}";
-    sha256 = "sha256-FT5AoqCHNf2sdKyejALOsL/zHrrxP7vdntagR9vA00I=";
+    sha256 = "sha256-Aeiq9IJReUxJUpeq5mSReK5foC5aY4fMSZli0ZUjYPc=";
   };
   # proxy vendor to avoid hash missmatches between linux and macOS
   proxyVendor = true;
-  vendorSha256 = "sha256-65bfTCMRJ8iL5ABGPqvkayw4zSn4KkCriEkWYa0Pe68=";
+  vendorSha256 = "sha256-vLbx/qmSZuteuvChnyA/wcEcIjU3zWkxSjfk8VBdgU4=";
 
-  subPackages = [ "cmd/otelcontribcol" ];
+  # there is a nested go.mod
+  sourceRoot = "source/cmd/otelcontribcol";
 
   # CGO_ENABLED=0 required for mac - "error: 'TARGET_OS_MAC' is not defined, evaluates to 0"
   # https://github.com/shirou/gopsutil/issues/976
   CGO_ENABLED = if stdenv.isLinux then 1 else 0;
```

Comment:
This bit looks interesting because the gopls dependency for m1 mac that was causing issues for us in non `-contrib` requires `GCO_ENABLED = 1`

Also I'm suprised we've not seen that aarch64-darwin failure here like non `-contrib`

Reply:
cc @uri-canva